src/utils/netutils.cpp

```diff
@@ -213,10 +213,55 @@ unique_fd ip_connect(const char* ip, int port, char const** error_result)
     int nbretry = 3;
     int retry_delai_ms = 1000;
     bool const no_log = false;
-    return connect_sck(sck, nbretry, retry_delai_ms, u.s, sizeof(u), text_target, no_log, error_result);
+    unique_fd client_sck = connect_sck(sck,
+				       nbretry,
+				       retry_delai_ms,
+				       u.s,
+				       sizeof(u),
+				       text_target,
+				       no_log,
+				       error_result);
+    
+    if (localIPAddress)
```

Comment:
candidate for a new function: `IpAddress get_client_address(fd, error)` or `[[nodiscard]] bool set_client_address(IpAddress, fd, error)`

```cpp
struct IpAddress
{
  char ip_addr[46]; // + static_assert(sizeof(IpAddress::ip_addr) >= INET6_ADDRSTRLEN)
};
```

Reply:
So, in rdp_params.hpp, do you suggest to replace "char client_address[46]"  by your "struct IpAddress" ?

Reply:
Rather like
```cpp
// create_module_rdp.cpp
IpAddress ip_addr;
if (!get_local_ip_address(ip_addr, sck.fd(), &error)) {
  // error
}
/* char const* */rdp_param.client_addr = ip_addr.c_str();
// ip_addr.c_str() is more expressive than ip_addr.ip_addr
```